src/openapi/streaming/subscription.js

```diff
@@ -163,6 +169,18 @@ function performAction(queuedAction) {
             }
             break;
 
+        case ACTION_UNSUBSCRIBE_BY_TAG_PENDING:
+            switch (this.currentState) {
+                case STATE_SUBSCRIBED:
+                case STATE_UNSUBSCRIBED:
+                    unsubscribeByTagPending.call(this);
+                    break;
+
+                default:
+                    log.error(LOG_AREA, 'unanticipated state', { state: this.currentState, action });

```

Comment:
why do we wait for subscriptions to be ready if er error unless the state is subscribed/unsubscirbed?

I was expecting that if you were unsubscribing, we ignore (double unsubscribe is no problem)
if you were subscribing, we wait until subscribed, and then change to ready for unsubscribe by tag

Reply:
Not sure I follow your question exactly but ... The only non-transitioning states are SUBSCRIBED and UNSUBSCRIBED, and we need to wait for all tagged subscriptions to be in one of these states. We need to handle both being subscribed and unsubscribed as some subscriptions could be unsubscribed but have the same tag as subscribed ones, we need to mark these as "ready" for unsubscribe by tag too.

I've added some logic in subscription-queue to handle unsubscribeByTag in a similar way to unsubscribe as I'd missed that.

Reply:
I understand now